src/utils/PluginDriver.ts

```diff
@@ -289,6 +300,23 @@ export class PluginDriver {
 		);
 	}
 
+	private getCompiledTransformFilter(filter: HookFilter): TransformHookFilter | undefined {
+		if (!this.compiledPluginFilters.transformFilter.has(filter)) {
+			this.compiledPluginFilters.transformFilter.set(
+				filter,
+				createFilterForTransform(filter.id, filter.code)
+			);
+		}
+		return this.compiledPluginFilters.transformFilter.get(filter);
```

Comment:
There is a common helper function `getOrCreate` that does this in a generic way. Then you can inline the two functions.